airflow/models/taskinstance.py

```diff
@@ -1120,10 +1125,33 @@ def handle_failure(self, error, test_mode=None, context=None, session=None):
             # only mark task instance as FAILED if the next task instance
             # try_number exceeds the max_tries.
             if self.is_eligible_to_retry():
-                self.state = State.UP_FOR_RETRY
-                self.log.info('Marking task as UP_FOR_RETRY')
-                if task.email_on_retry and task.email:
-                    self.email_alert(error)
+                if force_fail is not True:
```

Comment:
```suggestion
                if not force_fail:
```

or even better reverse the if and use `if force_fail:`